cfme/tests/cloud_infra_common/test_relationships.py

```diff
@@ -407,9 +407,12 @@ def test_change_network_security_groups_per_page_items(setup_provider, appliance
     """
     view = navigate_to(provider, "NetworkSecurityGroup")
     view.toolbar.view_selector.select('List View')
-    for item_count in [10, 20, 50]:
-        view.paginator.set_items_per_page(item_count)
-        assert len(view.entities.get_all()) <= view.paginator.items_per_page
+    view.paginator.set_items_per_page(1000)
+    total_count = len(view.entities.get_all())
```

Comment:
**Question**: 
1. Why are you setting ```set_items_per_page``` to 1000? if it is for getting total count then please check if there is any other way available to get total count of entities.

Reply:
We can refer "view.entities.get_all(surf_pages=True)"
to surf the page for entites to get total count.

Reply:
updated

Reply:
There's a method for this in `PaginationPane` - https://github.com/ManageIQ/integration_tests/blob/master/widgetastic_manageiq/__init__.py#L1809

It will be slow to use get_all if there are actually 1000 items. It'll also not get the total count if there are more than 1000.

Reply:
Thanks @mshriver!
updated with your suggestion